### DISCUSSION TITLE
How quickly should a high res webrtc or mse stream load in the frigate UI?

### DISCUSSION THREAD

**Discussion by dimatx** [blakeblackshear/frigate#6076] [Ask A Question] [answered]
Assuming everything is running smoothly and not resource constrained, how quickly should a highres (2688x1520 at 15 FPS) feed show on /cameras/<camera_name> web UI page? Mine seems to take between 1 and 3 seconds to get going. Wondering if that's to be expected or if it should be faster.

Using .12 with webrtc setup, and removed RTMP. H264 + G.711A for audio. Five total Dahua cameras with main stream for recording/streaming, and a lower res substream for detect.

**Comment by NickM-27** [comment]
The biggest factor will be iframe interval, the stream can only start on an iframe so that is a factor. Mine start basically instantly for webrtc and 1 second to start MSE

**Reply by dimatx** [reply]
Thanks @NickM-27. This is what mine looks like, do you think there's room for improvement from a perf standpoint?

![image](https://user-images.githubusercontent.com/24803941/232101893-4d1e03e9-1eb0-44f0-bcc8-47e2a368635b.png)

```
go2rtc:
  streams:
    garage:
      - rtsp://*****:*********@**********.com:554/cam/realmonitor?channel=1&subtype=0
```

**Reply by NickM-27** [reply, marked as answer]
With this config you will have an iframe every 2 seconds, so that lines up with what you are seeing. You may want to set it to 15 instead

**Reply by dimatx** [reply]
Immediate noticeable improvement. Thanks for the advice, Nick!

As an aside, I have noticed that even with the prior settings, my other setup that uses Scrypted for feeding data into HKSV does seem to start streaming a bit more quickly (or at least it feels that way). Any idea if we'd expect parity or if there's something that's happening over there that may use more resources to further reduce load times?

**Reply by NickM-27** [reply]
could be using active frame caching or any other possible thing to improve it, I wouldn't know
